examples/topics/stats/pyramid.py

```diff
@@ -0,0 +1,57 @@
+''' A pyramid plot showing the age distribution of the male and female Titanic passengers using the `Bokeh` hbar glyph.
+
+.. bokeh-example-metadata::
+    :sampledata: titanic
+    :apis: bokeh.plotting.figure.hbar
+    :refs: :ref:`ug_topics_stats_pyramid`
+    :keywords: pyramid, hbar
+'''
+import numpy as np
+
+from bokeh.models import CustomJSTickFormatter, Label
+from bokeh.plotting import figure, show
+from bokeh.sampledata.titanic import data as df
+
+sex_group = df.groupby("sex")
+
+female_ages = sex_group.get_group("female")["age"].dropna()
+male_ages = sex_group.get_group("male")["age"].dropna()
+
+bin_width = 5
+bins = np.arange(0, 72, bin_width)
+m_hist, edges = np.histogram(male_ages, bins=bins)
+f_hist, edges = np.histogram(female_ages, bins=bins)
+
+colors = ("#055BB2", "#CB6805", "#808080")
+bar_width = 4
+
+p = figure(title="Age population pyramid of titanic passengers, by gender", height=400, width=600,
+           x_range=(-90, 90), x_axis_label="count")
+
+p.hbar(right=m_hist * -1, y=edges[1:], height=bar_width, color=colors[0], line_width=0)
+
+p.hbar(right=f_hist, y=edges[1:], height=bar_width, color=colors[1], line_width=0)
+
+# add text to every other bar
+for i, (count, age) in enumerate(zip(f_hist, edges[1:])):
+    if i % 2 == 0:
+        p.text(x=count, y=edges[1:][i], text=[f"{age-bin_width}-{age}yrs"], x_offset=5, y_offset=7, text_font_size="12px", text_color=colors[2])
```

Comment:
the color for this text could maybe be one of the `DarkText` colors in the accessible palettes section? 

Also I do think this line is too long for an example. Those have to renderer in the docs so we can't go quite as far with line lengths. 

You could save an indentation level by doing 
```python
    if i % 2 == 1:
        continue
    p.text(....)
```
but i think the arguments are going to need a more vertical arrangement, regardless

Reply:
Thanks. it did occur to me to shorten that line somewhat but it eventually skipped my mind before I could get to it.